e2e/cephfs.go

```diff
@@ -848,6 +848,211 @@ var _ = Describe(cephfsType, func() {
 				}
 			})
 
+			By("Test subvolume snapshot and restored PVC metadata", func() {
+				err := createCephFSSnapshotClass(f)
+				if err != nil {
+					e2elog.Failf("failed to create CephFS snapshotclass: %v", err)
+				}
+				pvc, err := loadPVC(pvcPath)
+				if err != nil {
+					e2elog.Failf("failed to load PVC: %v", err)
+				}
+				pvc.Namespace = f.UniqueName
+				err = createPVCAndvalidatePV(f.ClientSet, pvc, deployTimeout)
+				if err != nil {
+					e2elog.Failf("failed to create PVC: %v", err)
+				}
+				snap := getSnapshot(snapshotPath)
+				snap.Namespace = f.UniqueName
+				snap.Spec.Source.PersistentVolumeClaimName = &pvc.Name
+				// create snapshot
+				snap.Name = f.UniqueName
+				err = createSnapshot(&snap, deployTimeout)
+				if err != nil {
+					e2elog.Failf("failed to create snapshot (%s): %v", snap.Name, err)
+				}
+				subvolumes, err := listCephFSSubVolumes(f, fileSystemName, subvolumegroup)
+				if err != nil {
+					e2elog.Failf("failed to list subvolumes: %v", err)
+				}
+				if len(subvolumes) == 0 {
+					e2elog.Failf("cephFS subvolumes list is empty %s", fileSystemName)
+				}
```

Comment:
subvolume name already present in the PV object just use it?

Reply:
I hope you mean
```
pvc, err := getPersistentVolumeClaim(c, pvc.Namespace, pvc.Name)
if err != nil {                                                 
    return nil, nil, fmt.Errorf("failed to get PVC: %w", err)   
}                                                               
```
And use `pvc.Spec.VolumeName` instead of `subvolumes[0].Name`

Reply:
volumeName is PV name, you need to get PV object and the subvolume name from pv.spec.csi.volumeattributes.